Xamarin.Essentials/DeviceDisplay/DeviceDisplay.mac.cs

```diff
@@ -0,0 +1,67 @@
+﻿using AppKit;
+using Foundation;
+
+namespace Xamarin.Essentials
+{
+    public static partial class DeviceDisplay
+    {
+        static uint keepScreenOnId = 0;
+        static NSObject screenMetricsObserver;
+
+        static bool PlatformKeepScreenOn
+        {
+            get
+            {
+                return keepScreenOnId != 0;
+            }
+
+            set
+            {
+                if (KeepScreenOn == value)
+                    return;
+
+                if (value)
+                {
+                    IOKit.PreventUserIdleDisplaySleep("KeepScreenOn", out keepScreenOnId);
+                }
+                else
+                {
+                    if (IOKit.AllowUserIdleDisplaySleep(keepScreenOnId))
+                        keepScreenOnId = 0;
+                }
+            }
+        }
+
+        static DisplayInfo GetMainDisplayInfo()
+        {
+            var frame = NSScreen.MainScreen.Frame;
+            var scale = NSScreen.MainScreen.BackingScaleFactor;
```

Comment:
I don't think this code is performance sensitive but, in general, you better avoid a roundtrip to ObjC and avoid a second call to the property, e.g.
```csharp
var main = NSScreen.MainScreen;
...
```
It's also more thread-safe to do so, e.g. it would be possible (pretty bad luck) than macOS main screen would change between the two calls, so the `frame` and `scale` would not represent the same. If you use a local then you can be sure both will represent the same screen.

Reply:
Thanks